[HIPIFY][clang][build][failure] Build failure against ToT LLVM 16.0.0git

**[Error]**
```cmd
src/HipifyAction.cpp:741:28: error: non-virtual member function marked 'override' hides virtual member function
                         ) override {
                           ^
```
**[LLVM]**
16.0.0git newer 2022.12.20

--- a/src/HipifyAction.cpp
+++ b/src/HipifyAction.cpp
@@ -730,8 +730,10 @@ public:
                           StringRef file_name, bool is_angled, clang::CharSourceRange filename_range,
 #if LLVM_VERSION_MAJOR < 15
                           const clang::FileEntry *file,
-#else
+#elif LLVM_VERSION_MAJOR == 15
                           Optional<clang::FileEntryRef> file,
+#else
+                          clang::OptionalFileEntryRef file,
 #endif
                           StringRef search_path, StringRef relative_path,
                           const clang::Module *imported

